rosbag2_storage/include/rosbag2_storage/storage_interfaces/base_write_interface.hpp

```diff
@@ -40,6 +41,8 @@ class ROSBAG2_STORAGE_PUBLIC BaseWriteInterface
 
   virtual void update_metadata(const BagMetadata & bag_metadata) = 0;
 
+  virtual void register_message_definition(const rosbag2_storage::MessageDefinition & message_definition) = 0;
```

Comment:
It's good that this is a struct type. I figure we will add the `type_description_interfaces::msg::TypeDescription` information eventually (or could even add it now?) so that storage implementation could do something with the inspectible description information.

Reply:
I was going to add `type_hash` after getting the bare minimum working, but i'm not sure what rosbag2 would want to do with TypeDescription content. I was going to retrieve just the raw definitions from the `GetTypeDescription` service and insert them here.